src/common/buffer.cc

```diff
@@ -2473,27 +2473,23 @@ void buffer::list::invalidate_crc()
 #include "common/ceph_crypto.h"
 using ceph::crypto::SHA1;
 
-boost::optional<sha1_digest_t> buffer::list::sha1()
+std::pair<bool, sha1_digest_t> buffer::list::sha1()
 {
   ptr nb;
-  unsigned pos = 0;
   unsigned char fingerprint[CEPH_CRYPTO_SHA1_DIGESTSIZE];
   if (_len == 0) {
-    return boost::none;
+    return make_pair(false, sha1_digest_t());
```

Comment:
could just return `{false, sha1_digest_t{}}` here. as template argument deduction in C++17 allows us to construct the appropriate type of `std::pair<>` instance without using `make_pair`.